nntrainer/tensor/cl_operations/blas_kernel_strings.h

```diff
@@ -106,11 +106,11 @@ static const std::string sgemm_cl_transAB_kernel_ =
     })";
 
 static const std::string addition_cl_kernel_ =
-  R"(__kernel void addition_cl(__global const float* input, __global float* output, const unsigned int size) {
+  R"(__kernel void addition_cl(const __global float* input, __global float* output, unsigned int size_input, unsigned int size_res) {
     #pragma printf_support
     size_t idx = get_global_id(0);
-    if (idx < size) {
-        output[idx] = output[idx] + input[idx];
+    if (idx < size_res) {
+        output[idx] = output[idx] + input[idx % size_input];
```

Comment:
You are assuming the input have multiple batch with same dimension... is it correct? `input[idx % size_input]` was written for the purpose (broadcast input with multi-batch). Do I correctly understand? If so, what about adding some condition to check whether `size_input` is multiple of `size_res`?

Reply:
It will always be a multiple. Broadcasting is done only for the case where inputA has batch n, and inputB has batch to be unity. This fulfils our requirement of running addition for MHA usecase.

Reply:
Oh, I see. I found the code checking the condition in `add_i_cl`.  However, what about adding codes to check its condition in `addition_cl` as well? The `addition_cl` might be used in other blas_cl_interface functions later. Isn't it? Please let me know if I missed something. BTW, there was an error in my previous comment. `size_res` should be multiple of `size_input` right? Sorry for the confusion.

Reply:
Yes right, size_res should be multiple of size_input, and this is already checked in add_i_cl. The addition_cl will be called via add_i_cl as far as I can understand. Could you clarify why another condition is needed in addition_cl?

Reply:
If it is called by the interface only, it doesn't need to be. If it is not, please consider it.